src/filters.ts

```diff
@@ -126,6 +126,37 @@ angular.module('3ema.filters', [])
     };
 })
 
+/**
+ * Convert mention elements to html elements
+ */
+.filter('mentionify', ['WebClientService', '$translate',
+    function (webClientService: WebClientService, $translate) {
+        return(text) => {
+            if (text !== null && text.length > 10) {
+                let result = text.match(/@\[([\*\@a-zA-Z0-9][\@a-zA-Z0-9]{7})\]/g);
+                if (result !== null) {
+                    result = ([...new Set(result)]);
+                    // Unique
+                    for (let possibleMention of result) {
+                        let identity = possibleMention.substr(2, 8);
+                        if (identity === '@@@@@@@@') {
+                            text = text.replace(
+                                new RegExp(possibleMention.replace(/([.*+?^=!:${}()|\[\]\/\\])/g, '\\$1'), 'g'),
```

Comment:
Whoa, I don't understand this RegEx. What does it do?

Couldn't we simply do `text.replace('@[@@@@@@@@]', <span class="mention all">' + $translate.instant('messenger.ALL') + '</span>')` first, and only then search for contact mentions?

Reply:
this saves one iteration, replace will be only called if one or more mentions (including the `all` mention) are found

Reply:
check my new commit